src/script_lua.c

```diff
@@ -838,10 +839,7 @@ static robj **luaArgsToRedisArgv(lua_State *lua, int *argc) {
 static int luaRedisGenericCommand(lua_State *lua, int raise_error) {
     int j;
     scriptRunCtx* rctx = luaGetFromRegistry(lua, REGISTRY_RUN_CTX_NAME);
-    if (!rctx) {
-        luaPushError(lua, "redis.call/pcall can only be called inside a script invocation");
```

Comment:
maybe we wanna keep that text as a comment (explaining the assertion)
something like `only supported inside script invocation, not referenced in the global 'redis' object during function library creation`

@MeirShpilraien do you remember why you used an error in the first place?
was it because it was reachable back then, or because we were paranoid about user somehow bypassing out guards and reaching these?

Reply:
But now there are 7 `serverAssert`, do we add 7 identical comments? :)

Reply:
if the comment is short (mine isn't) then i think it's ok.

another alternative is to suffice by the test checking this, and remove the assertions too.
p.s. the NPD crash that will happen if this ever happens is not that different than an assertion anyway.
i.e. as long as it was an error handling without the crash, it was arguably important. but now that it's an assert, i don't mind removing it.

Reply:
I prefer to keep `serverAssert`, because `scriptRunCtx` may only be used in a certain branch path, and using `serverAssert` we can find these undiscovered bugs much earlier.
WHat about only using comment `only supported inside script invocation`, feel that it's enough.

Reply:
Ok

Reply:
@oranagra it was reachable back then using some hacks (when you could change globals), but since we change the global table to be readonly it is not reachable anymore.
We also changed the test because it is reachable anymore: https://github.com/redis/redis/pull/10651/files#diff-49439b5bf4d47c8399638fe30445e0ebb24c45531aaa3c51632af37b35f75478R645

Theoretically if we have a bug or we missed something, it might still be reachable, but I guess that in this case an assert is a good idea, no?

Reply:
yes. i think assert is ok